src/testUtils.ts

```diff
@@ -242,7 +269,26 @@ function expandFilePathInOutput(output: string, cwd: string): string {
  */
 function targetArgs(testconfig: TestConfig): Thenable<Array<string>> {
 	if (testconfig.functions) {
-		return Promise.resolve([testconfig.isBenchmark ? '-bench' : '-run', util.format('^%s$', testconfig.functions.join('|'))]);
+		let params: string[] = [];
+		if (testconfig.isBenchmark) {
+			params = ['-bench', util.format('^%s$', testconfig.functions.join('|'))];
+		} else {
+			let testFunctions = testconfig.functions;
+			let testifyMethods = testFunctions.filter(isInstanceTestMethod);
+			if (testifyMethods.length > 0) {
+				// filter out testify methods
+				testFunctions = testFunctions.filter(fn => !isInstanceTestMethod(fn));
+				testifyMethods = testifyMethods.map(extractInstanceTestName);
+			}
+
+			if (testFunctions.length > 0) {
+				params = params.concat(['-run', util.format('^%s$', testFunctions.join('|'))]);
```

Comment:
We have to pass `-run ^$` even if `test.Functions` array is empty. Otherwise all tests will be run